aws/sns_test.go

```diff
@@ -0,0 +1,143 @@
+		Name: aws.String(name),
+	}
+
+	param := &sns.CreateTopicInput{
+		Name: testSNSTopic.Name,
+	}
+
+	// Do a coin-flip to choose either a FIFO or Standard SNS Topic
+	coin := []string{
+		"true",
+		"false",
+	}
+	rand.Seed(time.Now().UnixNano())
+	coinflip := coin[rand.Intn(len(coin))]
+	param.Attributes = make(map[string]string)
+	param.Attributes["FifoTopic"] = coinflip
+
+	// If we did choose to create a fifo queue, the name must end in ".fifo"
+	if coinflip == "true" {
+		param.Name = aws.String(fmt.Sprintf("%s.fifo", aws.StringValue(param.Name)))
+	}
+
+	output, err := svc.CreateTopic(context.TODO(), param)
+	if err != nil {
+		assert.Failf(t, "Could not create test SNS Topic: %s", errors.WithStackTrace(err).Error())
+	}
+
+	testSNSTopic.Arn = output.TopicArn
+
+	return testSNSTopic, nil
+}
+
+func TestListSNSTopics(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	},
+	)
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	snsTopicName := "aws-nuke-test-" + util.UniqueID()
+	testSNSTopic, createTestSNSTopicErr := createTestSNSTopic(t, session, snsTopicName)
+	require.NoError(t, createTestSNSTopicErr)
+	// clean up after this test
+	defer nukeAllSNSTopics(session, []*string{testSNSTopic.Arn})
+
+	snsTopicArns, err := getAllSNSTopics(session, time.Now(), config.Config{})
+	if err != nil {
+		assert.Fail(t, "Unable to fetch list of SNS Topics")
+	}
+
+	assert.Contains(t, awsgo.StringValueSlice(snsTopicArns), aws.StringValue(testSNSTopic.Arn))
+}
+
+func TestNukeSNSTopicOne(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	snsTopicName := "aws-nuke-test-" + util.UniqueID()
+	// We ignore errors in the delete call here, because it is intended to be a stop gap in case there is a bug in nuke.
+	testSNSTopic, createTestSNSTopicErr := createTestSNSTopic(t, session, snsTopicName)
+	require.NoError(t, createTestSNSTopicErr)
+
+	nukeErr := nukeAllSNSTopics(session, []*string{testSNSTopic.Arn})
+	require.NoError(t, nukeErr)
+
+	// Make sure the SNS Topic was deleted
+	snsTopicArns, err := getAllSNSTopics(session, time.Now(), config.Config{})
+	require.NoError(t, err)
+
+	assert.NotContains(t, aws.StringValueSlice(snsTopicArns), aws.StringValue(testSNSTopic.Arn))
+}
+
+func TestNukeSNSTopicMoreThanOne(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	testSNSTopicName := "aws-nuke-test-" + util.UniqueID()
+	testSNSTopicName2 := "aws-nuke-test-" + util.UniqueID()
+	// We ignore errors in the delete call here, because it is intended to be a stop gap in case there is a bug in nuke.
```

Comment:
This comment looks like it doesn't apply to these lines.

Reply:
Good catch - fixed in [bf21784](https://github.com/gruntwork-io/cloud-nuke/pull/354/commits/bf217849b3e8596bfd0306729ea7b43c33743114)